pkg/auth/dependency/sso/sso.go

```diff
@@ -105,11 +105,48 @@ type ProviderUserInfo struct {
 	Email string
 }
 
-// Provider defines SSO interface
-type Provider interface {
+type OAuthAuthorizationResponse struct {
+	Code  string
+	State string
+	Scope string
+	// Nonce is required when the underlying provider is OpenID connect compliant.
+	// The implementation is based on the suggestion in the spec.
+	// See https://openid.net/specs/openid-connect-core-1_0.html#NonceNotes
+	//
+	// The nonce is a cryptographically random string.
+	// The nonce passed to the provider is SHA256 hash of it.
+	// The get-authorization URL endpoint ensures the nonce in session cookie.
+	// The callback endpoint expect the user agent to include the nonce in session cookie.
+	// The nonce in session cookie will be validated against the hashed nonce in the ID token.
+	Nonce string
```

Comment:
We should also validate nonce on pure OAuth2 flow. ref: https://tools.ietf.org/html/draft-ietf-oauth-v2-threatmodel-06#section-5.3.5

Reply:
It requires the provider support. For now, we just validate nonce for OIDC provider.